### PR TITLE
Fix a bunch of typo

### PR DESCRIPTION
**Pull request by dzvon**
This PR will fix some typos detected by [typos].

I only picked the ones I was sure were spelling errors to fix, mostly in
the comments.

[typos]: https://github.com/crate-ci/typos

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1442,7 +1442,7 @@ Compatibility Notes
 - [Mixing Option and Result via `?` is no longer permitted in closures for inferred types.][86831]
 - [Previously unsound code is no longer permitted where different constructors in branches
   could require different lifetimes.][85574]
-- As previously mentioned the [`std::arch` instrinsics now uses stricter const checking][83278]
+- As previously mentioned the [`std::arch` intrinsic now uses stricter const checking][83278]
```
Comment (5225225) [inline RELEASES.md:1445]: `std::arch` isn't an intrinsic, it has intrinsics. The plural is correct here, but it was not previously spelled correctly

**Review by davidtwco — APPROVED** [review]

File: compiler/rustc_codegen_cranelift/src/intrinsics/mod.rs [hunk 2]
```diff
@@ -203,7 +203,7 @@ pub(crate) fn codegen_intrinsic_call<'tcx>(
             sym::transmute => {
                 crate::base::codegen_panic(fx, "Transmuting to uninhabited type.", source_info);
             }
-            _ => unimplemented!("unsupported instrinsic {}", intrinsic),
+            _ => unimplemented!("unsupported intrinsics {}", intrinsic),
```
Comment (5225225) [inline compiler/rustc_codegen_cranelift/src/intrinsics/mod.rs:206]: Plural incorrect here, this is a single intrinsic

**Comment by davidtwco** [comment]
r=me once @5225225's comment is addressed

File: library/core/src/slice/mod.rs [hunk 3]
```diff
@@ -2309,7 +2309,7 @@ impl<T> [T] {
     }
 
     /// Binary searches this slice for a given element.
-    /// This behaves similary to [`contains`] if this slice is sorted.
+    /// This behaves similar to [`contains`] if this slice is sorted.
```
Comment (5225225) [inline library/core/src/slice/mod.rs:2312]: probably meant to be `similarly`

File: library/std/src/sys/windows/cmath.rs [hunk 4]
```diff
@@ -44,7 +44,7 @@ mod shims {
 }
 
 // On 32-bit x86 MSVC these functions aren't defined, so we just define shims
-// which promote everything fo f64, perform the calculation, and then demote
+// which promote everything for f64, perform the calculation, and then demote
```
Comment (5225225) [inline library/std/src/sys/windows/cmath.rs:47]: `to`, probably?

File: compiler/rustc_codegen_cranelift/src/intrinsics/mod.rs (hunk 2 above)
Comment (dzvon) [reply compiler/rustc_codegen_cranelift/src/intrinsics/mod.rs]: Already corrected, thank you.

**Comment by davidtwco** [comment]
> Commit https://github.com/rust-lang/rust/commit/76d3a402c5200b4badff7d486273b46b78b78513 has been approved by `dawidtwco`

There's an irony that my name gets typo'd in this PR of all PRs.

**Comment by davidtwco** [comment]
We have a [no merge policy](https://rustc-dev-guide.rust-lang.org/git.html#no-merge-policy) - could you rebase instead?

**Comment by Dylan-DPC** [comment]
Are you sure you did it correctly? there are still conflicts to be resolved (you can see the conflicting files listed in the space below the CI checks)

**Comment by dzvon** [comment]
> Are you sure you did it correctly? there are still conflicts to be resolved (you can see the conflicting files listed in the space below the CI checks)

I know, the previous push was to remove that merge commit. I believe this time it works.

**Pull request merged**